packages/itwin/property-grid/src/hooks/UseNullValuesSetting.tsx

```diff
@@ -25,26 +26,33 @@ export function NullValueSettingContext({ children }: PropsWithChildren<{}>) {
 export function useNullValueSetting() {
   const [showNullValues, setShowNullValues] = useState(true);
   const { getShowNullValuesPreference, setShowNullValuesPreference } = useNullValueStorage();
+  const { onFeatureUsed } = useTelemetryContext();
+  const firstReportRef = useRef(false);
 
   // Get value from preferences storage
   useEffect(() => {
     void (async () => {
       const res = await getShowNullValuesPreference();
+      if (!firstReportRef.current) {
```

Comment:
I don't think we want to report only the first value. If preferences storage changes value will be updated and we should report that.

Reply:
Would it not update when the `updateShowNullValues` below is called? The reason I added the `firstReportRef` is that now it runs multiple times at the start and reports more than once.

Reply:
I guess it runs multiple times because of `React.StrictMode` rendering components twice for debugging purposes. (you can comment it out and it should report only once). `updateShowNullValues` will report when button is clicked. However, storage can be changed during runtime in which case setting value will change but won't be reported.

Reply:
Seems like you are right, removed the `firstReportRef`.